crates/node/src/command.rs

```diff
@@ -52,69 +52,138 @@ impl SubstrateCli for Cli {
-        utils::fetch_from_url(genesis_url, madara_path.clone() + "/configs/genesis-assets")?;
-    } else {
-        // TODO confirm with the CI that we are fetching all and fetch dynamically
-        // Issue #982
-        for file in constants::GENESIS_ASSETS_FILES {
-            let src_path = utils::get_project_path();
-            if let Ok(src_path) = src_path {
-                let src_path = src_path + "/configs/genesis-assets/" + file;
-                utils::copy_from_filesystem(src_path, madara_path.clone() + "/genesis-assets")?;
-            } else {
-                utils::fetch_from_url(
-                    constants::GENESIS_ASSETS_URL.to_string() + file,
-                    madara_path.clone() + "/genesis-assets",
+fn set_chain_spec(cli: &mut Cli) -> Result<(), String> {
+    // checks if it should retrieve and enable a specific chain-spec
+    let madara_path = cli.run.madara_path.clone().expect("Failed retrieving madara_path").to_str().unwrap().to_string();
+    let local_path = utils::get_project_path();
+
+    match (
+        cli.run.chain_spec_url.clone(),
+        cli.run.testnet,
+        cli.run.disable_url_fetch,
+        cli.run.disable_madara_configs,
+        local_path,
+    ) {
+        (Some(chain_spec_url), _, _, _, _) => {
+            utils::fetch_from_url(
+                chain_spec_url.clone(),
+                madara_path.clone() + "/chain-specs",
+                cli.run.update_configs,
+            )?;
+            let chain_spec = chain_spec_url.split('/').last().expect("Chain spec file name not found");
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/" + chain_spec);
+        }
+        (None, Some(Testnet::Sharingan), false, false, Ok(ref src_path)) => {
+            let src_path = src_path.clone() + "/configs/chain-specs/testnet-sharingan-raw.json";
+            utils::copy_from_filesystem(src_path, madara_path.clone() + "/chain-specs", cli.run.update_configs)?;
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
+        }
+        (None, Some(Testnet::Sharingan), false, false, _) => {
+            let madara_configs: configs::Configs =
+                serde_json::from_str(&utils::read_file_to_string(madara_path.clone() + "/configs/index.json")?)
+                    .expect("Failed to load madara configs");
+            // If testnet flag is specified, right now it's only Sharingan
+            for chain_spec in madara_configs.chain_specs {
+                configs::fetch_and_validate_file(
+                    madara_configs.remote_base_path.clone(),
+                    chain_spec,
+                    madara_path.clone() + "/configs/chain-specs/",
+                    cli.run.update_configs,
                 )?;
             }
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
         }
+        _ => {}
```

Comment:
So if either ` cli.run.disable_url_fetch` or `cli.run.disable_madara_configs` if `true` we will skip the two previous arms and go directly to this last "do nothing" arm?

Are you sure this is the behavior you want?

Reply:
No, I'm wrong

Reply:
Recheck the logic, pls

Reply:
```rust
if let (Some(Testnet::Sharingan), false) = (cli.run.testnet, cli.run.disable_madara_configs) {
   if let Some(ref src_path) = local_path {
     ...
   } else if !cli.run.disable_url_fetch {
     ...
   }
}
```

This is the same logic you wrote. I find it more readable this way.
Is this the logic you want this time?

Reply:
A lot of this logic only is relevant when we are running a sharingan testnet.
Can we extract it in a separated function?
I think it will make the code much clearer

```rust
if let (Some(Testnet::Sharingan) = cli.run.testnet {
   set_sharingan_chain_specs(cli);
}
```